test/extensions/problem.py

```diff
@@ -145,10 +145,10 @@ def forward_pass(self, sample_idx=None):
                 input, output, loss, each with batch axis first
         """
         if sample_idx is None:
-            input = self.input.clone().detach()
+            input = self.input.clone()
             target = self.target.clone().detach()
         else:
-            target = self.target.split(1, dim=0)[sample_idx].detach()
+            target = self.target.split(1, dim=0)[sample_idx]
             input = self.input.split(1, dim=0)[sample_idx].detach()
```

Comment:
Try removing the other `detach` ops, too.

Reply:
Worked.